TShockPluginManager/NugetCLI.cs

```diff
@@ -0,0 +1,207 @@
+/*
+TShock, a server mod for Terraria
+Copyright (C) 2022 Janet Blackquill
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+using System.CommandLine;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using NuGet.Packaging.Core;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace TShockPluginManager
+{
+	public static class NugetCLI
+	{
+		static public async Task<int> Main(List<string> args)
+		{
+			RootCommand root = new RootCommand(
+				description: "Manage plugins and their requirements"
+			);
+			Command cmdSync = new Command(
+				name: "sync",
+				description: "Install the plugins as specified in the plugins.json"
+			);
+			cmdSync.SetHandler(Sync);
+			root.Add(cmdSync);
+			return await root.InvokeAsync(args.ToArray());
+		}
+		class SyncManifest
+		{
+			[JsonPropertyName("packages")]
+			public Dictionary<string, NuGetVersion> Packages { get; set; } = new();
+			public PackageIdentity[] GetPackageIdentities() =>
+				Packages.Select((kvp) => new PackageIdentity(kvp.Key, kvp.Value))
+						.OrderBy(kvp => kvp.Id)
+						.ToArray();
+		}
+		public class NuGetVersionConverter : JsonConverter<NuGetVersion>
+		{
+			public override NuGetVersion? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+			{
+				return NuGetVersion.Parse(reader.GetString()!);
+			}
+
+			public override void Write(Utf8JsonWriter writer, NuGetVersion value, JsonSerializerOptions options)
+			{
+				writer.WriteStringValue(value.ToNormalizedString());
+			}
+		}
+		static async Task Sync()
+		{
+			var opts = new JsonSerializerOptions
+			{
+				ReadCommentHandling = JsonCommentHandling.Skip,
+				Converters =
+				{
+					new NuGetVersionConverter()
+				}
+			};
+			SyncManifest manifest;
+			try
+			{
+				string txt = await File.ReadAllTextAsync("packages.json");
+				manifest = JsonSerializer.Deserialize<SyncManifest>(txt, opts)!;
+			}
+			catch (System.IO.FileNotFoundException)
+			{
+				System.Console.WriteLine("You're trying to sync, but you don't have a packages.json file.");
+				System.Console.WriteLine("Without a list of plugins to install, I can't install any.");
```

Comment:
In general, TShock conventionally uses third person or indirect speech / tone when speaking to users. Can you adjust the language on these strings to be more cohesive with the rest of the project?

```suggestion
				System.Console.WriteLine("Unable to install plugins without a list of plugins to install. Please supply a packages.json file to continue.");
```

To be as precise as possible: we try to avoid having the software speak as if it is a human or otherwise an intelligent agent. "Is it okay to connect to the internet?" vs "May I connect to the internet?" (in particular, I think this doesn't flow very well into other languages -- but the primary hangup is that it's tone that doesn't really exist in the project elsewhere.)